### PR TITLE
Feature/all warnings as errors

### PR DESCRIPTION
**Pull request by qdsfdhvh**

### CODE REVIEW

File: common/gecko/sample/src/main/java/com/dimension/maskbook/common/gecko/sample/MainActivity.kt [hunk 1]
```diff
@@ -81,7 +81,7 @@ fun SampleApp(
     controller: WebContentController
 ) {
     val canGoBack by controller.canGoBack.collectAsState(initial = false)
-    val canGoForward by controller.canGoForward.collectAsState(initial = false)
+    // val canGoForward by controller.canGoForward.collectAsState(initial = false)
```
Comment (HuixingWong) [inline common/gecko/sample/src/main/java/com/dimension/maskbook/common/gecko/sample/MainActivity.kt:84]: Why not delete it?

File: wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/route/RegisterRoute.kt [hunk 2]
```diff
@@ -74,8 +74,8 @@ fun RegisterInit(
         onRecoveryAndSignIn = {
             navController.navigate(WalletRoute.Register.Recovery.Home)
         },
-        onSynchronization = {
-        },
+        // onSynchronization = {
```
Comment (HuixingWong) [inline wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/route/RegisterRoute.kt:77]: Why not delete it?

File: wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/ui/scenes/register/RegisterScene.kt [hunk 3]
```diff
@@ -48,7 +48,7 @@ import com.dimension.maskbook.wallet.R
 fun RegisterScene(
     onCreateIdentity: () -> Unit,
     onRecoveryAndSignIn: () -> Unit,
-    onSynchronization: () -> Unit,
+    // onSynchronization: () -> Unit,
```
Comment (HuixingWong) [inline wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/ui/scenes/register/RegisterScene.kt:51]: Why not delete it?

File: common/gecko/sample/src/main/java/com/dimension/maskbook/common/gecko/sample/MainActivity.kt (hunk 1 above)
Comment (qdsfdhvh) [reply common/gecko/sample/src/main/java/com/dimension/maskbook/common/gecko/sample/MainActivity.kt:84]: This code may be used for testing in sample, so just comment out it

File: wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/route/RegisterRoute.kt (hunk 2 above)
Comment (qdsfdhvh) [reply wallet/src/androidMain/kotlin/com/dimension/maskbook/wallet/route/RegisterRoute.kt:77]: This is useful but currently ui is hidden

**Review by HuixingWong — APPROVED** [review]
LGTM!

**Pull request merged**